spec/facter/query_parser_spec.rb

```diff
@@ -79,6 +79,25 @@
         contain_exactly(an_instance_of(Facter::SearchedFact).and(having_attributes(fact_class: path_class)))
     end
 
+    context 'when user query contains metacharacters (like dots)' do
+      it 'finds the structured fact and not the alias' do
+        query_list = ['ldom.domainname.impl']
+        ldom_class = 'Facter::Macosx::Ldom'
```

Comment:
I think this class only exists in my PR? That said, I'm not 100% sure how to stub a loaded fact that only exists on a single platform (unlike `os.name`)

Reply:
Oh shoot, good catch

Reply:
I think I can just set it to Facter::Solaris::Ldom instead